#Baseline #3

#Number of orders: 1
#Number of shipments per order: 1
#Challenges:
#  -Interruption: no
#  -Drop parts: no
#  -Sensor blackout: no
#  -Part flipping: yes
#  -Faulty parts: yes
#  -Moving obstacle: no
#Containers
# -Belt: no
# -Bins: yes
# -Shelf: yes
#AGV
# -order_0:
#   -shipment_0: agv2
#Time limit: Yes

time_limit: 500

options:
  gazebo_state_logging: true
  insert_models_over_bins: true
  insert_models_over_shelves: true
  belt_population_cycles: 0
  model_type_aliases:
    order_part1: pulley_part_red
    order_part2: disk_part_green
    order_part3: disk_part_blue

shelf_layout:
  row_1: ['base','collar',0,'base'] #shelves: 3, 4, 5
  row_2: ['pipe', 0,'base','base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11


orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 1
    destinations: [agv2]
    products:
      part_0:
        type: pulley_part_red
        pose:
          xyz: [-0.12, -0.2, 0]
          rpy: [0, 0, 0]
      part_1:
        type: pulley_part_red
        pose:
          xyz: [0.12, 0.2, 0]
          rpy: ['pi', 0, 0]
      part_2:
        type: disk_part_green
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: [0, 0, 0]
      part_3:
        type: disk_part_blue
        pose:
          xyz: [-0.15, 0.2, 0]
          rpy: [0, 0, 0]
      part_4:
        type: disk_part_blue
        pose:
          xyz: [0.15, -0.2, 0]
          rpy: [0, 0, 0]


models_over_bins:
  bin13:
    models:
      disk_part_blue:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3
  bin16:
    models:
      disk_part_green:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3
  bin3:
    models:
      pulley_part_red:
        xyz_start: [0.15, 0.15, 0.0]
        xyz_end: [0.45, 0.45, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 2
        num_models_y: 1

models_over_shelves:
  shelf5:
    models:
      pulley_part_red:
        xyz_start: [1, 0.4, 0.0]
        xyz_end: [0., 0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 2
        num_models_y: 1

#Faulty products
faulty_products:
  - disk_part_blue_1
  - disk_part_green_5
